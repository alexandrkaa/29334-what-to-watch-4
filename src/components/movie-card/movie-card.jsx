import React from 'react';
import PropTypes from 'prop-types';
import withVideoPlayer from '../../hocs/with-video-player/with-video-player.js';

const MovieCard = (props) => {
  const {movie, onMovieTitleClick} = props;
  const {title} = movie;
  const _onMovieCardMouseEnter = () => {
    props.onPlay();
  };

  const _onMovieCardMouseLeave = () => {
    props.onPause();
  };

  const _onMovieTitleClick = onMovieTitleClick.bind(null, movie);
  return (
    <article className="small-movie-card catalog__movies-card" onMouseEnter={_onMovieCardMouseEnter} onMouseLeave={_onMovieCardMouseLeave} >
      <div className="small-movie-card__image" onClick={_onMovieTitleClick}>
        {props.children}
      </div>
      <h3 className="small-movie-card__title" onClick={_onMovieTitleClick}>
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >{title}</a>
      </h3>
    </article>
  );
};

MovieCard.propTypes = {
  movie: PropTypes.shape({
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    movieBackground: PropTypes.string.isRequired,
    movieGenre: PropTypes.string.isRequired,
    movieDate: PropTypes.string.isRequired,
    movieImage: PropTypes.string.isRequired,
    movieRatingScore: PropTypes.string.isRequired,
    movieRatingLevel: PropTypes.string.isRequired,
    movieRatingCount: PropTypes.string.isRequired,
    movieDirector: PropTypes.string.isRequired,
    movieStarring: PropTypes.string.isRequired,
    movieDescription: PropTypes.arrayOf(
        PropTypes.string.isRequired
    ),
    moviePreview: PropTypes.string.isRequired
  }),
  onMovieTitleClick: PropTypes.func.isRequired,
  onPlay: PropTypes.func.isRequired,
  onPause: PropTypes.func.isRequired,
  children: PropTypes.element.isRequired
};
export {MovieCard};
export default withVideoPlayer(MovieCard);
