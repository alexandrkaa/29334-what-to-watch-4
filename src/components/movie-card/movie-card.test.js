import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {MovieCard} from "./movie-card.jsx";

const movie = {
  id: 4,
  image: `img/we-need-to-talk-about-kevin.jpg`,
  movieBackground: `img/bg-the-grand-budapest-hotel.jpg`,
  movieDate: `2000`,
  movieDescription: `In the 1930s, the Grand Budapest Hotel is a populaâ€¦y boy, becomes Gustave&apos;s friend and protege.`,
  movieDirector: `Wes Anderson`,
  movieGenre: `Thriller`,
  movieImage: `img/the-grand-budapest-hotel-poster.jpg`,
  movieRatingCount: `349 ratings`,
  movieRatingLevel: `Bad`,
  movieRatingScore: `3`,
  movieStarring: [`Jude Law`, `Willem Dafoe`],
  title: `Dardjeeling Limited`,
  moviePreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  backgroundColor: `#000000`,
  movieVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isFavorite: false,
};

it(`<MovieCard /> should render Big Bang Theory movie`, () => {
  const onPlay = jest.fn();
  const onPause = jest.fn();
  const onMovieTitleClick = jest.fn();
  const children = <></>;
  const tree = renderer
    .create(
        <BrowserRouter>
          <MovieCard
            key={movie.id}
            movie={movie}
            onPlay={onPlay}
            onPause={onPause}
            onMovieTitleClick={onMovieTitleClick}
            history={{}}
          >
            {children}
          </MovieCard>
        </BrowserRouter>,
        {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
