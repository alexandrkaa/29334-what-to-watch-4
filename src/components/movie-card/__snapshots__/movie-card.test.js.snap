// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard /> should render Big Bang Theory movie 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
>
  <div
    className="small-movie-card__image"
    onClick={[Function]}
  >
    <img
      alt="Dardjeeling Limited"
      height="175"
      src="img/we-need-to-talk-about-kevin.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Dardjeeling Limited
    </a>
  </h3>
</article>
`;
